packages/zudoku/src/vite/plugin-docs.ts

```diff
@@ -29,37 +33,49 @@ const viteDocsPlugin = (getConfig: () => ZudokuPluginOptions): Plugin => {
           config.mode === "internal"
             ? `import { markdownPlugin } from "${config.moduleDir}/src/lib/plugins/markdown/index.tsx";`
             : `import { markdownPlugin } from "zudoku/plugins/markdown";`,
-          `const configuredDocsPlugins = [];`,
+          `const docsPluginOptions = [];`,
         ];
 
-        const docsConfigs: ZudokuDocsConfig[] = config.docs
-          ? Array.isArray(config.docs)
-            ? config.docs
-            : [config.docs]
-          : [{ files: DEFAULT_DOCS_FILES }];
+        const resolver = new DocResolver(config);
+        const docsConfigs = resolver.getDocsConfigs();
+
+        for (let i = 0; i < docsConfigs.length; i++) {
+          const docsConfig = docsConfigs[i];
+
+          if (!docsConfig) continue;
+
+          const globbedFiles = await glob(docsConfig.files, {
+            root: config.rootDir,
+            ignore: ["**/node_modules/**", "**/dist/**", "**/.git/**"],
+            absolute: false,
+          });
 
-        docsConfigs.forEach((docsConfig) => {
           code.push(
-            ...[
-              `// @ts-ignore`, // To make tests pass
-              `const markdownFiles = import.meta.glob(${JSON.stringify(docsConfig.files)}, {`,
-              `  eager: false,`,
-              `});`,
-              `configuredDocsPlugins.push(markdownPlugin({ `,
-              ` markdownFiles,`,
-              ` defaultOptions: ${JSON.stringify(docsConfig.defaultOptions)},`,
-              ` filesPath: ${JSON.stringify(docsConfig.files)}`,
-              `}));`,
-            ],
+            `const fileImports${i} = Object.assign({`,
+            ...globbedFiles.map(
+              (file) =>
+                `  "${ensureLeadingSlash(file)}": () => import("${ensureLeadingSlash(path.join(config.basePath ?? "", file))}"),`,
+            ),
+            `});`,
```

Comment:
Following scenario:
- Root dir is `/path/to/docs`
- The markdown docs config is `/docs/**/*.md`
- The basePath config is set to `/docs`

When these circumstances come together `import.meta.glob` does not work properly:

> Internal server error: Failed to resolve import "/articles/accounts/billing.md" from "virtual:zudoku-docs-plugins". Does the file exist?

The generated + evaluated output of the plugin looks like:

```
const fileImports0 = /* #__PURE__ */ Object.assign({
  "/docs/articles/accounts/billing.md": () => import("/docs/articles/accounts/billing.md")
  // ...
})
```

However to correctly `import()` it needs the base path prepended. This PR does exactly that, this seems to work for all scenarios.
It also makes the generated code from the plugin more transparent as it doesn't need to be evaluated in the browser.